pkg/sql/catalog/descpb/structured.proto

```diff
@@ -1365,6 +1365,10 @@ message TypeDescriptor {
     // Represents a special multi-region enum type which tracks available regions
     // as its enum values.
     MULTIREGION_ENUM = 2;
+    // Represents the virtual record type created on behalf of each table. This
+    // kind of TypeDescriptor is *never* persisted to disk! If you are here,
+    // thinking about using or persisting this value, you should *not* do that!
+    VIRTUAL_TABLE_RECORD_TYPE = 3;
```

Comment:
I don't hate it, but "virtual" may be a bit overloaded. What about "transient" instead? or "implicit"?